pkgs/development/cuda-modules/setup-hooks/mark-for-cudatoolkit-root/default.nix

```diff
@@ -0,0 +1,7 @@
+# Internal hook, used by cudatoolkit and cuda redist packages
+# to accommodate automatic CUDAToolkit_ROOT construction
+{ makeSetupHook }:
+makeSetupHook {
+  name = "mark-for-cudatoolkit-root";
+  substitutions.logFromSetupHook = ../utilities/log-from-setup-hook.sh;
```

Comment:
Wouldn't `log-from-setup-hook` be more reusable if it were a setup hook?

Reply:
I had tried this as an earlier approach -- I frequently got errors about the log bash function not being found, and couldn't get it sourced correctly or reliably. I had also tried making it a derivation using the `writeShellApplication` builder and including it as a native dependency so I could call it as a program from within bash, but had the same issues.

The only reliable way I found to ensure it was in scope was to use a substitution and manually source it.

Reply:
Interesting. I would expect `autoFixElfFiles` adding `logFromSetupHook` to its `propagatedBuildInputs` to be sufficient. I wonder why that fails

Reply:
Moved to #302694